terminology.md

```diff
@@ -247,6 +247,59 @@ In Firefox's Web Console the [command line interpreter](https://developer.mozill
 - [REPL](https://en.wikipedia.org/wiki/Read%E2%80%93eval%E2%80%93print_loop)
 .....
 
+### Agent
+
+#### Definition
+An Agent is a thread that holds all state within a JavaScript VM instance.
+
+### Realm
+#### Definition
+A realm consists of a set of [intrinsic](###Intrinsic) objects, an ECMAScript global environment, all of the ECMAScript code that is loaded within the scope of that global environment, and other associated state and resources (i.e. a global object and an associated set of [primordial](###Primoridal) objects). Today, in the browser, realms can be created via same origin iframes.
+#### References
+[ECMA262 Spec](https://tc39.es/ecma262/#sec-code-realms)
+
+### Intrinsic
+#### Definition
+All objects that can be observed by a program, whether it is via a global ref, or some other mechanism to access those objects, and then those that can be accessed via syntax/grammar. A reference such as %name% means the intrinsic object.
```

Comment:
I don't think this is quite right; e.g. %AsyncFromSyncIteratorPrototype% is an intrinsic that specifically claims to be unreachable from ECMAScript code (and even if it fails in that, there's still a semantic distinction). Per vocabulary shared by @caridy and I think @erights, _primordials_ are the subset of intrinsics that are reachable from code (and further, _undeniables_ are the strict subset of primordials that can be reached after arbitrary manipulation of named bindings—e.g., `[].__proto__` to get %ArrayPrototype%).

Reply:
(The definition also appears to include objects created by the program itself, which certainly aren't intrinsics.)

Reply:
Referencing the spec, the closest definition I can find for `intrinsic` is (paraphrasing a bit):
>  built-in objects that are explicitly referenced by the algorithms of the ECMA262 specification.

I'm fine with that definition though I wouldn't mind both of your help fleshing it out a bit 😄